### ISSUE TITLE
convert onnx success but run error

### ISSUE THREAD

**Issue by cqray1990** [onnx/onnx#4339]
# Bug Report

### Is the issue related to model conversion?
If the ONNX checker reports issues with this model then this is most probably related to the converter used to convert the original framework model to ONNX. Please create this bug in the appropriate converter's GitHub repo (pytorch, tensorflow-onnx, sklearn-onnx, keras-onnx, onnxmltools) to get the best help.

### Describe the bug
Please describe the bug clearly and concisely

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*): 16.04
- ONNX version (*e.g. 1.7*):  1.8
- Python version:3.6.8
- GCC/Compiler version (if compiling from source):
- CMake version:
- Protobuf version:
- Visual Studio version (if applicable):

export onnx:

```
  dynamic_axes = {"input": {2: "height", 3: "width"},"output": {2: "height", 3: "width"}}
   torch.onnx.export(model.model,
                          x,
                          onnx_file_name,
                          export_params=True,
                          verbose=True,
                          opset_version=11,
                          do_constant_folding=True,
                          input_names=input_names, output_names=output_names,
                          dynamic_axes=dynamic_axes)
i check onnxmodel no errors:
import onnx
model  = onnx.load("./mm_dy.onnx")
onnx.checker.check_model(model)
onnx.helper.printable_graph(model.graph)
```

but when i run onnx model raise error:

`onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Add node. Name:'' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/math/element_wise_ops.h:479 void onnxruntime::BroadcastIterator::Init(int64_t, int64_t) axis == 1 || axis == largest was false. Attempting to broadcast an axis by a dimension other than 1. 40 by 44`

finally i compare my code and find if  use follows code,the error raise

```
    def _upsample_add(self, x, y):
        _, _, H, W = map(int, y.size())
        return F.interpolate(x, size=(H,W),mode="nearest") + y
```

but when use follows code: there is no errors , why????

```
    def _upsample_add(self, x, y):
        return F.interpolate(x, size=y.size()[2:]) + y
```

**Comment by cqray1990** [comment]
@jcwchen @huitseeker @ezyang @shinh

**Comment by jcwchen** [comment]
Hi @cqray1990,
Does this model pass by onnx.checker with full_check? Could you please provide this model for me to repro this issue if possible? Thanks.

**Comment by cqray1990** [comment]
> hers is export onx success, if i change as follows to export: it raises errors
def _upsample_add(self, x, y):
    _, _, H, W = map(int, y.size())
    return F.interpolate(x, size=(H,W),mode="nearest") + y


[net.onnx.zip](https://github.com/onnx/onnx/files/9155737/net.onnx.zip)

**Comment by jcwchen** [comment]
Thanks for providing the model. However, ONNX Runtime runs fine with it from my end (I used [1,3,2,3] as input shape). Could you please provide the code that you ran with ONNX Runtime?

**Comment by cqray1990** [comment]
> Thanks for providing the model. However, ONNX Runtime runs fine with it from my end (I used [1,3,2,3] as input shape). Could you please provide the code that you ran with ONNX Runtime?

the mode i send for u can success run, but i change 
the _upsample_add function

def _upsample_add(self, x, y):
    return F.interpolate(x, size=y.size()[2:]) + y

as 

def _upsample_add(self, x, y):
_, _, H, W = map(int, y.size())
return F.interpolate(x, size=(H,W),mode="nearest") + y

then  it can't be export onnx model

**Comment by daquexian** [comment]
> > Thanks for providing the model. However, ONNX Runtime runs fine with it from my end (I used [1,3,2,3] as input shape). Could you please provide the code that you ran with ONNX Runtime?
> 
> the mode i send for u can success run, but i change the _upsample_add function
> 
> def _upsample_add(self, x, y): return F.interpolate(x, size=y.size()[2:]) + y
> 
> as
> 
> def _upsample_add(self, x, y): _, _, H, W = map(int, y.size()) return F.interpolate(x, size=(H,W),mode="nearest") + y
> 
> then it can't be export onnx model, did u read my problem clearly?

It makes no sense to send the "success" model to us. We can help you only if you send the problematic model.

**Comment by jcwchen** [comment]
Yes, please provide the failed model, which will be very helpful for us to investigate the problem. Actually, I feel it's more like a question for the PyTorch-ONNX exporter, since those 2 functions seem to be identical to me and I am not sure why the exporter convert them differently. I would suggest you raise this issue in the exporter repo instead (in PyTorch repo and mention ONNX) to get the best help from the converter team.
